src/composables/multisigAccounts.ts

```diff
@@ -0,0 +1,294 @@
+  return storedMultisig ? JSON.parse(storedMultisig) : [];
+}
+
+const multisigAccounts = ref(<IMultisigAccount[]>[]);
+const activeMultisigAccountId = ref('');
+const isMultisigDashboard = ref(false);
+
+const initPollingWatcher = createPollingBasedOnMountedComponents();
+
+export function useMultisigAccounts({ store }: UseMultisigOptions) {
+  const { getDrySdk, getSdk } = useSdk({ store });
+
+  const accounts = computed<IAccount[]>(() => store.getters.accounts);
+  const activeNetwork = computed<INetwork>(() => store.getters.activeNetwork);
+
+  multisigAccounts.value = getStoredMultisigAccounts(activeNetwork.value.networkId);
+
+  const multisigProgress = ref<IMultisigCreationStep>(MULTISIG_CREATION_STEPS.preparing);
+
+  /**
+   * @param {number} noOfConfirmations no of minimum confirmation needed to process a transaction
+   * @param {string[]} signers
+   */
+  async function deployMultisigAccount(
+    noOfConfirmations: number,
+    signersAddresses: [string, string, ...string[]],
+  ): Promise<IMultisigAccountBase> {
+    if (noOfConfirmations > signersAddresses.length) throw Error('Number of confirmations exceed amount of signers');
+    multisigProgress.value = MULTISIG_CREATION_STEPS.preparing;
+
+    const sdk = await getSdk();
+    const drySdk = await getDrySdk();
+
+    // create a temporary account
+    const gaAccount = Crypto.generateKeyPair();
+
+    const multisigContractInstance = await drySdk.getContractInstance({
+      source: multisigContract,
+    });
+    await multisigContractInstance.compile();
+    multisigProgress.value = MULTISIG_CREATION_STEPS.compiled;
+
+    const contractArgs = [noOfConfirmations, signersAddresses];
+    // Build Attach transaction
+    const attachTX = await drySdk.gaAttachTx({
+      ownerId: gaAccount.publicKey,
+      code: multisigContractInstance.bytecode,
+      callData: multisigContractInstance.calldata.encode(multisigContractInstance._name, 'init', contractArgs),
+      authFun: Crypto.hash('authorize'),
+      gas: await multisigContractInstance._estimateGas('init', contractArgs),
+      options: {
+        innerTx: true,
+      },
+    });
+    const { rawTx } = await drySdk.send(attachTX.tx, {
+      innerTx: true,
+      onAccount: gaAccount,
+    });
+    // Submit transaction using the default account
+    await sdk.payForTransaction(rawTx, {
+      waitMined: true,
+      modal: false,
+    });
+    multisigProgress.value = MULTISIG_CREATION_STEPS.deployed;
+
+    const gaContract = await drySdk.getAccount(gaAccount.publicKey);
+    multisigProgress.value = MULTISIG_CREATION_STEPS.created;
+
+    return {
+      contractId: gaContract.contractId,
+      gaAccountId: gaAccount.publicKey,
+    };
+  }
+
+  async function getTransactionByHash(txHash: string) {
+    return fetchJson(`${activeNetwork.value.multisigBackendUrl}/tx/${txHash}`);
+  }
+
+  function setActiveGaAccountId(gaAccountId: string) {
+    activeMultisigAccountId.value = gaAccountId;
```

Comment:
Maybe it would make sense to stick with one name: `...gaAccount` or `...multisigAccount`.